units/src/internal_macros.rs

```diff
@@ -18,7 +18,22 @@
 /// You must specify `$other_ty` and you may not use `Self`. So e.g. you need
 /// to write `impl ops::Add<Amount> for Amount { ... }` when calling this macro.
 macro_rules! impl_op_for_references {
+    ($(
+        impl $($op_trait:ident)::+<$other_ty:ty> for $ty:ty {
+            $($whatever:tt)*
+        }
+    )*) => {
+        $($crate::internal_macros::impl_op_for_references! {
+            single
+            impl $($op_trait)::+<$other_ty> for $ty {
+                $($whatever)*
+            }
+        })*
+    };
```

Comment:
Why not just wrap the entire thing in `$()+` at parameters and at body?

Reply:
I was trying to reduce the line noise on the "real" part of the implementation. I had an idea to put another layer of recursion in where I'd also handle generics, and a layer where I'd turn `Self` into `$ty`, but I got hung up fighting the grammar.

In the end getting rid of the `$(` and `)+` was all I managed so it might not have been worth it. I don't mind changing it to your way.

Reply:
Rebased and changed as suggested.

Reply:
Thanks! Unfortunately, I don't have energy to review it today. Hopefully tomorrow.